mycpp/marksweep_heap.h

```diff
@@ -0,0 +1,44 @@
+#ifndef MARKSWEEP_H
+#define MARKSWEEP_H
+
+#include <new>
+#include <unordered_set>
+
+typedef uint64_t u64;
+
+class MarkSweepHeap {
+ public:
+  MarkSweepHeap() {
+  }
+  void Init(int space_size);
+
+  void* Allocate(int);
+
+  void MarkAllReferences(Obj* obj);
+
+  void PushRoot(Obj** p) {
+    assert(roots_top_ < kMaxRoots);
+    roots_[roots_top_++] = p;
+  }
+
+  void PopRoot() {
+    roots_top_--;
+  }
+
+  void Collect(int to_space_size = 0);
+
+  int roots_top_;
+  Obj** roots_[kMaxRoots];  // These are pointers to Obj* pointers
+
+  u64 current_heap_bytes;
+  u64 max_heap_bytes;
+
+  // TODO(Jesse): This should really be in an 'internal' build
+  bool is_initialized_ =
+      true;  // mark/sweep heap doesn't need to be initialized
+
+  std::unordered_set<void*> AllAllocations;
+  std::unordered_set<void*> MarkedAllocations;
```

Comment:
Please following the existing style of using member names like:

```
marked_allocations_
max_heap_bytes_
```

i.e. lower_case_ with trailing underscore

Reply:
Fixed